src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java

```diff
@@ -57,11 +59,16 @@ public HttpClientWebRequest(CloseableHttpClient httpClient, HttpClientContext ht
    * Releases the connection by Closing.
    */
   @Override
-  public void close() {
-    if (null != httpPost) {
+  public void close() throws IOException {
+    // First check if we can close the response, by consuming the complete response
+    // This releases the connection but keeps it alive for future requests
+    // If that is not possible, we simply cleanup the whole connection
+    if (response != null && response.getEntity() != null) {
+      EntityUtils.consume(response.getEntity());
+    } else if (httpPost != null) {
       httpPost.releaseConnection();
-      //postMethod.abort();
     }
+
     httpPost = null;
```

Comment:
Should this be inside the elseIf?  i.e. if you just consume the response should be leave the httpPost as is rather than setting it to null?